packages/api/src/cli/storage.ts

```diff
@@ -0,0 +1,297 @@
+      }
+
+      return a.source === search.source;
+    });
+
+    return res === undefined ? false : res;
+  }
+
+  setIdentifier(identifier: string) {
+    this.identifier = identifier;
+  }
+
+  /**
+   * Determine if the current spec + identifier we're working with is already in the lockfile.
+   */
+  isInLockfile() {
+    return Boolean(this.getFromLockfile());
+  }
+
+  /**
+   * Retrieve the lockfile record for the current spec + identifier if it exists in the lockfile.
+   */
+  getFromLockfile() {
+    const lockfile = Storage.getLockfile();
+    return lockfile.apis.find(a => a.identifier === this.identifier);
+  }
+
+  getIdentifierStorageDir() {
+    if (!this.isInLockfile()) {
+      throw new Error(`${this.source} has not been saved to storage yet and must do so before being retrieved.`);
+    }
+
+    return path.join(Storage.getAPIsDir(), this.identifier);
+  }
+
+  getAPIDefinition() {
+    const file = fs.readFileSync(path.join(this.getIdentifierStorageDir(), 'openapi.json'), 'utf8');
+
+    return JSON.parse(file);
+  }
+
+  saveSourceFiles(files: Record<string, string>) {
+    if (!this.isInLockfile()) {
+      throw new Error(`${this.source} has not been saved to storage yet and must do so before being retrieved.`);
+    }
+
+    return new Promise(resolve => {
+      const savedSource: string[] = [];
+      Object.entries(files).forEach(([fileName, contents]) => {
+        const sourceFilePath = path.join(this.getIdentifierStorageDir(), fileName);
+
+        fs.writeFileSync(sourceFilePath, contents);
+
+        savedSource.push(sourceFilePath);
+      });
+
+      resolve(savedSource);
+    });
+  }
+
+  async load() {
+    return this.fetcher.load().then(async spec => this.save(spec));
+  }
+
+  /**
+   * @example <caption>Storage directory structure</caption>
+   * .api/
+   * ├── api.json             // The `package-lock.json` equivalent that records everything that's
+   * |                        // installed, when it was installed, what the original source was,
+   * |                        // and what version of `api` was used.
+   * └── apis/
+   *     ├── readme/
+   *     |   ├── node_modules/
+   *     │   ├── index.js     // We may offer the option to export a raw TS file for folks who want
+   *     |   |                // that, but for now it'll be a compiled JS file.
+   *     │   ├── types.ts     // All types for their SDK, ready to use in an IDE.
+   *     │   |── openapi.json
+   *     │   └── package.json
+   *     └── petstore/
+   *         ├── node_modules/
+   *         ├── index.js
+   *         ├── types.ts
+   *         ├── openapi.json
+   *         └── package.json
+   *
+   * @param spec
+   */
+  save(spec: OASDocument) {
+    if (!this.identifier) {
+      throw new TypeError('An identifier must be set before saving the API definition into storage.');
+    }
+
+    // Create our main `.api/` directory.
+    if (!fs.existsSync(Storage.dir)) {
```

Comment:
None that I'm aware of. I'm mostly using `*Sync` in a few spots in this file because the methods they're in are nested within some other promises and it's easier to use the sync version of `fs` then having to deal with nested promises or promises in a foreach.